subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c

```diff
@@ -56,6 +69,6 @@ int nrf_cloud_jwt_generate(uint32_t time_valid_s, char *const jwt_buf, size_t jw
 		return err;
 	}
 
-	return 0;
+	return (modem_time_is_valid ? 0 : ETIME);
```

Comment:
> I don't particularly like that you would need to call some function twice

Yes I agree.  
As with any function, the caller is responsible for checking the return value and acting appropriately.  This is just a convenience function, there is nothing stopping a user from using `modem_jwt_generate` directly without specifying an exp time.
I can add text in the function header to indicate that using a JWT without an expiration time could have security risks.
Also, I can add a LOG_WRN to indicate that the exp claim was not populated.

Reply:
I discussed this more with TommiK and I think the choice between current implementation and what TommiK suggests (have dedicated value for `time_valid_s` to indicate whether expiration time is added or not to JWT) depends on how we intend this to be used.

Which way the customers behave:
1) They decide whether they require expiration time or not and provide this information into the function. JWT will the have the expiration time or not based on what was requested.
2) They decide that let's add expiration time but if JWT with it fails, we are ok to use it without.

And an unrelated question maybe to @jayteemo. We are generating a lot of JWT's as we do it for every request. If there is no expiration time in these requests, will all these JWT's work with the cloud forever although we would use them just once? I guess they would work.

Reply:
Once I have answers/opinions to above questions. We can discuss whether some of the behavior should be configurable as I'm not sure how other libraries could use this. They certainly cannot silently generate JWT and acceppt them without expiration time but I'm not sure if they should fail their operations either in such a case.

Reply:
If you generate a JWT with the same parameters elit will produce the same value.

Reply:
My concern is that someone calls the function with a validity time, but accidentally uses a JWT which never expires because the modem did not have valid time. A warning about this in the function description might be sufficient as well.

I'm just not sure whether using a JWT without validity time is ever going to be a good idea. This was pretty much the whole point of using run-time generated JWTs vs. compile time keys, right?

The device should have two possible time sources, LTE network and NTP. If network connectivity doesn't work, it doesn't matter whether we can generate a JWT or not, because the REST request can't anyway be sent. So, the only possibility would be that modem doesn't get time from the LTE network and NTP server can't be reached/server is down.

Reply:
> I'm just not sure whether using a JWT without validity time is ever going to be a good idea.

AT%JWT allows it to be done.  We can discourage it, but we can't prevent it.

Reply:
Security issues are often things that developers let go in the default path because people don't fully understand all the details, and security is kind of additional burden that doesn't increase the functionality of the product. This is not to say it wouldn't be an important thing.

Hence, there is a big difference if our default implementation generates a JWT without validity time unless developer does additional checks (in this case checks the return value). Compared to an implementation which would fail for such a JWT. While the difference is kind of small, I think we give a different message to the developers also.

Having said this, we need to decide if missing modem time will:
1) Still generate JWT and give indication that this is without JWT (current implementation in the PR)
2) A parameter indicates whether JWT should be created without validity time (Tommi's suggestion to use specific value for validity time)
3) Always fail if JWT doesn't contain validity time. Customer can still use modem_jwt_generate() or AT%JWT but we send clear message that JWT without validity time shouldn't be used unless customer knows what they are doing.

Reply:
I'm starting to lean towards option 3. There's still two options if somebody wants to generate such JWTs. I don't think we need to or should offer such option in `nrf_cloud_jwt_generate()` convenience function.

Reply:
i will update to option 3